lib/fluent/plugin/in_forward.rb

```diff
@@ -125,7 +125,12 @@ def on_message(msg, chunk_size, source)
         return
       end
 
-      # TODO format error
+      # TODO: raise an exception if broken chunk is generated by recoverable situation
+      unless msg.is_a?(Array)
+        log.warn "incoming chunk is broken:", source: source, msg: msg
```

Comment:
`msg` should be `inspect` -ed for various values, ex: strings with escape code or others.